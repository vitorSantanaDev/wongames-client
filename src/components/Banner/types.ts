import { RibbonColors, RibbonSizes } from 'components/Ribbon/types'

export type BannerProps = {
  img: string
  title: string
  subtitle: string
  buttonLabel: string
  buttonLink: string
  ribbon?: React.ReactNode
  ribbonSize?: RibbonSizes
  ribbonColor?: RibbonColors
}
