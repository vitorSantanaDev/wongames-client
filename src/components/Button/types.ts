export type ButtonProps = {
  icon?: React.ReactNode
  fullWidth?: boolean
  children?: React.ReactNode
  size?: 'small' | 'medium' | 'large'
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void
}
