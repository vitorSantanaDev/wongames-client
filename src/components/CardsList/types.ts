import { PaymentCard } from 'components/PaymentOptions/types'

export type CardsListProps = {
  cards?: PaymentCard[]
}
