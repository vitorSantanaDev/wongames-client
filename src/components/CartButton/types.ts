import { ButtonProps } from 'components/Button/types'

export type CartButtonProps = {
  id: string
  hasText?: boolean
} & Pick<ButtonProps, 'size'>
