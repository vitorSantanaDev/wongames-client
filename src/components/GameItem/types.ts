export type GameItemProps = {
  id: string
  img: string
  title: string
  price: string
  downloadLink?: string
  paymentInfo?: PaymentInfoProps
}

export type PaymentInfoProps = {
  flag: string
  image: string
  number: string
  purchasedDate: string
}
