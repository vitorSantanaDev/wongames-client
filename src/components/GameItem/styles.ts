import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.small};
    border-bottom: 0.1rem solid ${theme.colors.lightGray};

    ${media.greaterThan('medium')`
      display: flex;
    `}
  `}
`

export const GameContent = styled.div`
  ${() => css`
    display: flex;
  `}
`

export const ImageBox = styled.div`
  ${({ theme }) => css`
    flex-shrink: 0;
    margin-right: 1.2rem;
    width: 9.6rem;
    height: 5.6rem;

    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }

    ${media.greaterThan('medium')`
      width: 15rem;
      height: 7rem;
      margin-right: ${theme.spacings.xsmall};
    `}
  `}
`

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
`

export const Title = styled.h3`
  ${({ theme }) => css`
    color: ${theme.colors.black};
    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.small};
    line-height: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.xxsmall};

    ${media.greaterThan('medium')`
      font-size: 2rem;
      line-height: 2rem;
    `}
  `}
`

export const Price = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-weight: ${theme.font.bold};
    padding: 0.2rem ${theme.spacings.xxsmall};
    font-size: ${theme.font.sizes.small};
    border-radius: ${theme.border.radius};
    background-color: ${theme.colors.secondary};
  `}
`

export const DownloadLink = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
    margin-left: ${theme.spacings.xxsmall};
  `}
`

export const PaymentContent = styled.div`
  ${({ theme }) => css`
    min-width: 28rem;

    color: ${theme.colors.gray};
    font-size: ${theme.font.sizes.small};

    display: flex;
    flex-direction: column;

    margin-top: ${theme.spacings.xsmall};

    ${media.greaterThan('medium')`
      flex: 1;
      margin-top: 0;
      flex-direction: column-reverse;
      justify-content: space-between;
      align-items: flex-end;
    `}
  `}
`

export const CardInfo = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;

    img {
      margin-left: ${theme.spacings.xxsmall};
    }

    ${media.lessThan('medium')`
      margin-top: ${theme.spacings.xsmall};
    `}
  `}
`
