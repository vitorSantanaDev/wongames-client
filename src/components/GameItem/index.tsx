import { Download } from '@styled-icons/boxicons-solid'
import { GameItemProps } from './types'

import * as S from './styles'

const GameItem = ({
  image,
  price,
  title,
  paymentInfo,
  downloadLink
}: GameItemProps) => (
  <S.Wrapper>
    <S.GameContent>
      <S.ImageBox>
        {/*eslint-disable-next-line @next/next/no-img-element*/}
        <img src={image} alt={title} />
      </S.ImageBox>
      <S.Content>
        <S.Title>
          {title}
          {!!downloadLink && (
            <S.DownloadLink
              href={downloadLink}
              target="_blank"
              aria-label={`Get ${title} here`}
            >
              <Download size={22} />
            </S.DownloadLink>
          )}
        </S.Title>
        <S.Price>{price}</S.Price>
      </S.Content>
    </S.GameContent>
    {!!paymentInfo && (
      <S.PaymentContent>
        <div>{paymentInfo.purchasedDate}</div>
        <S.CardInfo>
          <span>{paymentInfo.number}</span>
          {/*eslint-disable-next-line @next/next/no-img-element*/}
          <img src={paymentInfo.image} alt={paymentInfo.flag} />
        </S.CardInfo>
      </S.PaymentContent>
    )}
  </S.Wrapper>
)

export default GameItem
