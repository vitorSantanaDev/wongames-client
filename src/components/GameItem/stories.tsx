import { Story, Meta } from '@storybook/react'
import GameItem from '.'
import { GameItemProps } from './types'

export default {
  title: 'GameItem',
  component: GameItem,
  args: {
    image:
      'https://images.hdqwalls.com/download/cyberpunk-biker-gang-4k-cf-1360x768.jpg',
    price: 'R$ 215,00',
    title: 'Red Dead Redemption 2'
  }
} as Meta

export const Default: Story<GameItemProps> = (args) => <GameItem {...args} />

export const WithPayment: Story<GameItemProps> = (args) => (
  <GameItem {...args} />
)

WithPayment.args = {
  downloadLink: 'http://link',
  paymentInfo: {
    flag: 'mastercard',
    image: '/img/Master.png',
    number: '**** **** **** 4326',
    purchasedDate: 'Purchased made on 07/20/2023'
  }
}
