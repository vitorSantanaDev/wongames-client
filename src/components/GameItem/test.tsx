import { GameItemProps, PaymentInfoProps } from './types'

import GameItem from '.'
import { render, screen } from 'utils/test-utils'

const props: GameItemProps = {
  image:
    'https://images.hdqwalls.com/download/cyberpunk-biker-gang-4k-cf-1360x768.jpg',
  price: 'R$ 215,00',
  title: 'Red Dead Redemption 2'
}

describe('<GameItem />', () => {
  it('should render the item', () => {
    const { container } = render(<GameItem {...props} />)

    const itemHeading = screen.getByRole('heading', {
      name: /Red Dead Redemption 2/i
    })
    expect(itemHeading).toBeInTheDocument()

    const itemPrice = screen.getByText('R$ 215,00')
    expect(itemPrice).toBeInTheDocument()

    const itemImage = screen.getByRole('img', {
      name: /Red Dead Redemption 2/i
    })
    expect(itemImage).toHaveAttribute('src', props.image)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the item with download link', () => {
    const downloadLink = 'https://link'
    render(<GameItem {...props} downloadLink={downloadLink} />)

    const link = screen.getByRole('link', { name: `Get ${props.title} here` })
    expect(link).toHaveAttribute('href', downloadLink)
  })

  it('should render the the payment info', () => {
    const paymentInfo: PaymentInfoProps = {
      flag: 'mastercard',
      image: '/img/Master.png',
      number: '**** **** **** 4326',
      purchasedDate: 'Purchased made on 07/20/2023'
    }
    render(<GameItem {...props} paymentInfo={paymentInfo} />)

    const flagImage = screen.getByRole('img', { name: paymentInfo.flag })
    expect(flagImage).toHaveAttribute('src', paymentInfo.image)

    const cardNumber = screen.getByText(paymentInfo.number)
    expect(cardNumber).toBeInTheDocument()

    const purchasedDate = screen.getByText(paymentInfo.purchasedDate)
    expect(purchasedDate).toBeInTheDocument()
  })
})
