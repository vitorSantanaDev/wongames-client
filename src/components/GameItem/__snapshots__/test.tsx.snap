// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameItem /> should render the item 1`] = `
.c0 {
  padding: 2.4rem;
  border-bottom: 0.1rem solid #EAEAEA;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
}

.c2 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c4 {
  color: #030517;
  font-weight: 600;
  font-size: 1.4rem;
  line-height: 1.4rem;
  margin-bottom: 0.8rem;
}

.c5 {
  color: #FAFAFA;
  font-weight: 600;
  padding: 0.2rem 0.8rem;
  font-size: 1.4rem;
  border-radius: 0.4rem;
  background-color: #3CD3C1;
}

@media (min-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c2 {
    width: 15rem;
    height: 7rem;
    margin-right: 1.6rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <img
        alt="Red Dead Redemption 2"
        src="https://images.hdqwalls.com/download/cyberpunk-biker-gang-4k-cf-1360x768.jpg"
      />
    </div>
    <div
      class="c3"
    >
      <h3
        class="c4"
      >
        Red Dead Redemption 2
      </h3>
      <p
        class="c5"
      >
        R$ 215,00
      </p>
    </div>
  </div>
</div>
`
